WooCommerce/WooCommerceTests/POS/Controllers/PointOfSaleItemsControllerTests.swift

```diff
@@ -125,6 +125,7 @@ final class PointOfSaleItemsControllerTests {
         let initialItems = MockPointOfSaleItemService.makeInitialItems()
         itemProvider.items = initialItems
         itemProvider.shouldSimulateTwoPages = true
+        await sut.loadInitialItems()
```

Comment:
🗒️ for reviewer: `sut.loadInitialItems()` was added in a few test cases before `sut.loadNextItems()`, because the latest page number tracking has been moved from the items controller to the pagination tracker `AsyncPaginationTracker`. Before this move, the page number is already 1 by default in the items controller without actually having fetched the first page. When calling `loadNextItems`, it increments the page number to 2 and the mock item provider previously appends the page 2 items to the page 1 items and returns all these items for page 2 request:

https://github.com/woocommerce/woocommerce-ios/blob/931208508fe47dfffccd537a36698abd8d07c50f/WooCommerce/WooCommerceTests/POS/Mocks/MockPOSItemProvider.swift#L34-L36

This behavior doesn't seem expected, as the item provider just returns the items for a given page number without any state from previous pages. This PR removes this behavior, but please let me know if I misunderstood the expected behavior of item provider.

Reply:
Yeah that seems better